dvc/render/converter/vega.py

```diff
@@ -221,14 +222,24 @@ def flat_datapoints(self, revision):  # noqa: C901, PLR0912
             )
 
         all_datapoints = []
-        all_y_fields = {y_field for _, y_field in ys}
+        if ys:
+            all_y_files, all_y_fields = list(zip(*ys))
+            all_y_fields = set(all_y_fields)
+            all_y_files = set(all_y_files)
+        else:
+            all_y_files = set()
+            all_y_fields = set()
 
         # override to unified y field name if there are different y fields
         if len(all_y_fields) > 1:
             props_update["y"] = "dvc_inferred_y_value"
         else:
             props_update["y"] = first(all_y_fields)
 
+        # get common prefix to drop from field and file names
+        common_field_prefix = _get_common_prefix(all_y_fields)
```

Comment:
To make sure I get it right, `field` would be the name actually stored inside the plot, right?

I see the point for `filename`, but why also alter `field`? 
Was the motivation coming from DVCLive storing the full paths in `field`?

Reply:
> Was the motivation coming from DVCLive storing the full paths in `field`?

Yes, at least partially, although a little hard to remember at this point 😅 . It still could be long if people use field names like `test/acc` on their own 🤔 .

Now that you bring it up, it's no longer an issue for DVCLive, and it's probably better to not mess with the user-provided field name. I think it makes sense, so I can update to drop altering the field.

Reply:
> Now that you bring it up, it's no longer an issue for DVCLive, and it's probably better to not mess with the user-provided field name. I think it makes sense, so I can update to drop altering the field.

Yep, that was where I was heading to 👍

Reply:
I figured 😄 . Should be updated with those changes now.